Makefile

```diff
@@ -22,38 +22,58 @@ export GOSUMDB ?= sum.golang.org
 GIT_COMMIT  = $(shell git rev-list -1 HEAD)
 GIT_VERSION = $(shell git describe --always --abbrev=7 --dirty)
 # By default, disable CGO_ENABLED. See the details on https://golang.org/cmd/cgo
-CGO         ?= 0
+CGO  ?= 0
+PWD  := $(shell pwd)
 
 LOCAL_ARCH := $(shell uname -m)
 ifeq ($(LOCAL_ARCH),x86_64)
-	TARGET_ARCH_LOCAL=amd64
+  TARGET_ARCH_LOCAL=amd64
 else ifeq ($(shell echo $(LOCAL_ARCH) | head -c 5),armv8)
-	TARGET_ARCH_LOCAL=arm64
+  TARGET_ARCH_LOCAL=arm64
 else ifeq ($(shell echo $(LOCAL_ARCH) | head -c 4),armv)
-	TARGET_ARCH_LOCAL=arm
+  TARGET_ARCH_LOCAL=arm
 else
-	TARGET_ARCH_LOCAL=amd64
+  TARGET_ARCH_LOCAL=amd64
 endif
 export GOARCH ?= $(TARGET_ARCH_LOCAL)
 
 LOCAL_OS := $(shell uname)
 ifeq ($(LOCAL_OS),Linux)
-   TARGET_OS_LOCAL = linux
+  TARGET_OS_LOCAL = linux
 else ifeq ($(LOCAL_OS),Darwin)
-   TARGET_OS_LOCAL = darwin
+  TARGET_OS_LOCAL = darwin
 else
-   TARGET_OS_LOCAL ?= windows
+  TARGET_OS_LOCAL ?= windows
 endif
 export GOOS ?= $(TARGET_OS_LOCAL)
 
 ifeq ($(GOOS),windows)
-BINARY_EXT_LOCAL:=.exe
-GOLANGCI_LINT:=golangci-lint.exe
-# Workaround for https://github.com/golang/go/issues/40795
-BUILDMODE:=-buildmode=exe
+  FINDBIN := where
+  BINARY_EXT_LOCAL:=.exe
+  GOLANGCI_LINT:=golangci-lint.exe
+  # Workaround for https://github.com/golang/go/issues/40795
+  BUILDMODE:=-buildmode=exe
 else
-BINARY_EXT_LOCAL:=
-GOLANGCI_LINT:=golangci-lint
+  FINDBIN := which
+  BINARY_EXT_LOCAL:=
+  GOLANGCI_LINT:=golangci-lint
+endif
+
+# Install golangci_lint if not installed
+GH_LINT_VERSION := $(shell grep 'GOLANGCI_LINT_VER:' .github/workflows/components-contrib.yml | xargs | cut -d" " -f2)
+
+ifeq (, $(shell $(FINDBIN) golangci-lint))
+  $(info [*] golangci_lint not installed)
+  $(info [*] Installing now...)
+  $(shell curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell go env GOPATH)/bin $(GH_LINT_VERSION))
```

Comment:
While I appreciate the intention here, I am not sure I would feel comfortable with a "curl | sh" being executed automatically. Say what you want about them (I personally am not against the practice on its own), but in this context it doesn't feel right: a "make lint" command is expected to run a linter and doing a "curl | sh" in there is at best mischievous IMHO. 

Perhaps rather than executing the command we could just print on the terminal something like `Please run curl .... | sh ...` and instruct the user to do that? Or simply post a link to the official website.

Reply:
How about add check target like "golangci-verify":
```
golangci-verify:
  # check bin, install if not existed
lint: golangci-verify
```

Reply:
@ItalyPaleAle yeah that makes sense and would tend to agree with you. Let me take a look at this again